### PR TITLE
etcdserver: range queries with min/max create revision

### PR DESCRIPTION
**Pull request by heyitsanthony**
More API updates pursuant to #5412 / reducing sychronization message overhead.

### CODE REVIEW

File: etcdserver/etcdserverpb/rpc.proto [hunk 1]
```diff
@@ -390,6 +390,15 @@ message RangeRequest {
...
   int64 max_mod_revision = 11;
+
+  // min_create_revision is the lower bound for returned key create revisions; all keys with
+  // lesser create trevisions will be filtered away.
+  int64 min_create_revision = 12;
+
+  // max_create_revision is the upper bound for returned key create revisions; all keys with
+  // greater create revisions will be filtered away.
+  int64 max_create_revision = 13;
+
```
Comment (xiang90) [inline etcdserver/etcdserverpb/rpc.proto]: remove extra space

File: clientv3/concurrency/key.go [hunk 2]
```diff
@@ -46,32 +45,21 @@ func waitDelete(ctx context.Context, client *v3.Client, key string, rev int64) e
...
-	for err == nil {
-		for len(resp.Kvs) > 0 {
-			i := len(resp.Kvs) - 1
-			if resp.Kvs[i].CreateRevision <= maxRev {
-				break
-			}
-			resp.Kvs = resp.Kvs[:i]
+// waitDeletes efficiently waits until all keys matching the prefix and no greater
+// than the create revision.
+func waitDeletes(ctx context.Context, client *v3.Client, pfx string, maxRev int64) error {
```
Comment (xiang90) [inline clientv3/concurrency/key.go]: maxRev -> maxCreateRev

File: clientv3/concurrency/key.go [hunk 3]
```diff
@@ -46,32 +45,21 @@ func waitDelete(ctx context.Context, client *v3.Client, key string, rev int64) e
...
+	getOpts := append(v3.WithLastCreate(), v3.WithMaxCreateRev(maxRev))
```
Comment (xiang90) [inline clientv3/concurrency/key.go]: should we still add v3.WithPrefix option?

**Comment by xiang90** [comment]
LGTM

File: clientv3/concurrency/key.go (hunk 3 above)
Comment (heyitsanthony) [inline clientv3/concurrency/key.go]: that's included with `WithLastCreate`

Comment (xiang90) [inline clientv3/concurrency/key.go]: ok. got it.

**Comment by xiang90** [comment]
@heyitsanthony Shall we get this merged?

**Pull request merged**